lib/subliminal/providers/opensubtitles.py

```diff
@@ -199,6 +199,12 @@ def query(self, languages, hash=None, size=None, imdb_id=None, query=None, seaso
             filename = subtitle_item['SubFileName']
             encoding = subtitle_item.get('SubEncoding') or None
 
+            # When searching by IMDB, provider discard query, season and episode
+            # https://trac.opensubtitles.org/projects/opensubtitles/wiki/XMLRPC
+            # "If you define imdbid, then moviehash, moviebytesize and query is ignored"
+            if not (series_season == int(season) and series_episode == int(episode)):
+                continue
+
```

Comment:
with this fix i get only correct season + episode results.
without i got 247 results!!

![image](https://cloud.githubusercontent.com/assets/2620870/19008851/b0d3d9f2-8744-11e6-9396-61b2d2c2d502.png)